### PR TITLE
[Localization] Updated Spanish translations

### PR DESCRIPTION
**Pull request by GoldTra**
## What are the changes?
- Updated egg.ts
- Updated biome.ts
- Updated menu.ts
- Updated starter-select-ui-handler.ts
- Updated voucher.ts
- Updated modifier-type.ts
- Updated tutorial.ts 

## Why am I doing these changes?
I decided to introduce these changes to improve the overall user experience, particularly for Spanish-speaking players who may not understand English well.
I've changed the tutorial as indicated in issue (ES) #1517.

## What did change?
Translated all descriptions of the items.
Translated all official Pokémon item names.
Translated the hatching process using annotations from the egg status.
Translated vouchers and gacha.
Revised the translation of evolution to align with the official Spanish translation used in Pokémon games.
Updated biome translations based on feedback from the Spanish community on Discord.

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by torranx** [comment]
Would it be possible to include this on your PR?
https://github.com/pagefaultgames/pokerogue/issues/1517

**Comment by GoldTra** [comment]
> Would it be possible to include this on your PR (at least the ES)? #1517

tutorial.ts file and settings are already translated #1583

**Comment by GoldTra** [comment]
> Would it be possible to include this on your PR (at least the ES)? #1517

I have translated the tutorial according to your issue, and the settings are translated in PR #1583.

**Comment by torranx** [comment]
> > Would it be possible to include this on your PR (at least the ES)? #1517
> 
> I have translated the tutorial according to your issue, and the settings are translated in PR #1583.

thanks!

File: src/locales/es/egg.ts [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
+  "egg": "Huevo",
+  "greatTier": "Raro",
+  "ultraTier": "Épico",
+  "masterTier": "Legendario",
+  "defaultTier": "Común",
+  "hatchWavesMessageSoon": "Se escuchan sonidos. ¡Pronto saldrá!",
+  "hatchWavesMessageClose": "A veces se mueve. Debe estar a punto de salir.",
+  "hatchWavesMessageNotClose": "¿Qué habrá dentro? Tendrás que esperar un poco más.",
+  "hatchWavesMessageLongTime": "Parece que a este huevo le va a costar mucho abrirse.",
+  "gachaTypeLegendary": "Tasa aumentada en Legendario",
```
Comment (InnocentGameDev) [inline src/locales/es/egg.ts:13]: I think it'd be better to translate as "mayor tasa de X" o "mayor probabilidad de X", which more closely resembles the original text and is free of an extra verb/participle.

File: src/locales/es/egg.ts [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
+  "gachaTypeMove": "Tasa aumentada en Movimiento Huevo",
```
Comment (InnocentGameDev) [inline src/locales/es/egg.ts:14]: This should emphasize that's its the RARE eggmoves that are more likely -- you always get some eggmove, but it's the rare ones that are more likely.

File: src/locales/es/menu.ts [hunk 3]
```diff
@@ -36,11 +36,11 @@ export const menu: SimpleTranslationEntries = {
...
   "boy": "Chico",
   "girl": "Chica",
-  "evolving": "¿Uh?\n¡{{pokemonName}} está evolucionando!",
-  "stoppedEvolving": "{{pokemonName}} no ha evolucionado.",
+  "evolving": "¡Anda!\n¡{{pokemonName}} está evolucionando!",
+  "stoppedEvolving": "¿Eh?\n¡La evolución de {{pokemonName}} se ha detenido!",
   "pauseEvolutionsQuestion": "¿Quieres detener las evoluciones de {{pokemonName}}?\nSiempre pueden ser reactivadas desde la pantalla de tu equipo.",
   "evolutionsPaused": "Se han detenido las evoluciones para {{pokemonName}}.",
-  "evolutionDone": "¡Felicidades!\n¡Tu {{pokemonName}} ha evolucionado a {{evolvedPokemonName}}!",
+  "evolutionDone": "¡Enhorabuena!\n¡Tu {{pokemonName}} ha evolucionado a {{evolvedPokemonName}}!",
```
Comment (InnocentGameDev) [inline src/locales/es/menu.ts:43]: Why are we changing this? I used the same wording as the English file here.

File: src/locales/es/biome.ts [hunk 4]
```diff
@@ -2,19 +2,19 @@ import { SimpleTranslationEntries } from "#app/plugins/i18n";
...
   "FOREST": "Bosque",
   "SEA": "Mar",
   "SWAMP": "Pantano",
   "BEACH": "Playa",
   "LAKE": "Lago",
-  "SEABED": "Fondo del mar",
+  "SEABED": "Lecho marino",
   "MOUNTAIN": "Montaña",
-  "BADLANDS": "Badlands",
+  "BADLANDS": "Meseta",
```
Comment (InnocentGameDev) [inline src/locales/es/biome.ts:17]: "Meseta" means plateau, which is not what we're going for here. I'd say "Tierras baldías" is a good enough translation

**Review by InnocentGameDev — CHANGES_REQUESTED** [review]
Added some feedback and some questions on things I disagree with to some extent (like further changes in menu.ts)

File: src/locales/es/egg.ts (hunk 1 above)
Comment (GoldTra) [reply src/locales/es/egg.ts:13]: I'm going to change

File: src/locales/es/menu.ts (hunk 3 above)
Comment (GoldTra) [reply src/locales/es/menu.ts:43]: I have changed the translation to match the wording in the official translation.

File: src/locales/es/biome.ts (hunk 4 above)
Comment (GoldTra) [reply src/locales/es/biome.ts:17]: I am making the change you suggested for badlands.

File: src/locales/es/egg.ts (hunk 1 above)
Comment (GoldTra) [reply src/locales/es/egg.ts:13]: I am making the change you suggested

**Comment by GoldTra** [comment]
> Added some feedback and some questions on things I disagree with to some extent (like further changes in menu.ts)

Changed!!!

**Comment by GoldTra** [comment]
> Everything added so far is correct and a welcome addition to the game for Spanish users! Thank you!
> 
> I'm unsure about this, but I think we are still missing the usage of "V" explanation in the stat-changes in tutorial.ts .

I updated with "V" explanation

**Review by InnocentGameDev — APPROVED** [review]
Seems great! Thank you!

File: src/locales/es/modifier-type.ts [hunk 5]
```diff
@@ -4,357 +4,357 @@ export const modifierType: ModifierTypeTranslationEntries = {
...
-    "BATON": { name: "Baton", description: "Allows passing along effects when switching Pokémon, which also bypasses traps" },
+    "BATON": { name: "Baton", description: "Permite pasar los efectos al cambiar de Pokémon, también evita las trampas" },
 
-    "SHINY_CHARM": { name: "Shiny Charm", description: "Dramatically increases the chance of a wild Pokémon being Shiny" },
-    "ABILITY_CHARM": { name: "Ability Charm", description: "Dramatically increases the chance of a wild Pokémon having a Hidden Ability" },
+    "SHINY_CHARM": { name: "Amuleto Iris", description: "Aumenta drásticamente la posibilidad de que un Pokémon salvaje sea Shiny" },
+    "ABILITY_CHARM": { name: "Amuleto Habilidad", description: "Aumenta drásticamente la posibilidad de que un Pokémon salvaje tenga una habilidad oculta" },
 
-    "IV_SCANNER": { name: "IV Scanner", description: "Allows scanning the IVs of wild Pokémon. 2 IVs are revealed per stack. The best IVs are shown first" },
+    "IV_SCANNER": { name: "Escáner IV", description: "Permite escanear los IV de Pokémon salvajes. Se revelan 2 IV por cada objeto Los mejores IV se muestran primero" },
```
Comment (InnocentGameDev) [inline src/locales/es/modifier-type.ts:217]: "por cada acumulación.", "objeto" is fine too but add the missing "."

**Review by Tempo-anon — APPROVED** [review]

File: src/locales/es/modifier-type.ts [hunk 6]
```diff
@@ -4,357 +4,357 @@ export const modifierType: ModifierTypeTranslationEntries = {
...
-    "x_speed": "X Speed",
-    "x_accuracy": "X Accuracy",
-    "dire_hit": "Dire Hit",
+    "x_attack": "Ataque X",
+    "x_defense": "Defensa X",
+    "x_sp_atk": "Ataq. Esp. X",
+    "x_sp_def": "Def. Esp. X",
+    "x_speed": "Velocidad X",
+    "x_accuracy": "Precisión X",
+    "dire_hit": "Directo",
```
Comment (InnocentGameDev) [inline src/locales/es/modifier-type.ts:244]: Name is "Crítico X" in generation IX

**Comment by InnocentGameDev** [comment]
@GoldTra I'll make a PR with the typos that didn't make it to the merge

File: src/locales/es/modifier-type.ts (hunk 5 above)
Comment (GoldTra) [reply src/locales/es/modifier-type.ts:217]: I didnt add the "." bc all description didnt have it

Comment (InnocentGameDev) [reply src/locales/es/modifier-type.ts:217]: In this case, the "." I'm referring to is in the middle of the description, to separate the two sentences:

"Allows scanning the IVs of wild Pokémon(HERE). (HERE) 2 IVs are revealed per stack. The best IVs are shown first"

Comment (InnocentGameDev) [reply src/locales/es/modifier-type.ts:217]: It'd be missing this "." -> ...per stack (HERE).(HERE) The best IVs...

**Pull request merged**